packages/runtime/runtime-definitions/src/dataStoreContext.ts

```diff
@@ -606,6 +609,26 @@ export interface IFluidDataStoreContext extends IFluidParentContext {
 	 * and its children with the GC details from the previous summary.
 	 */
 	getBaseGCDetails(): Promise<IGarbageCollectionDetailsBase>;
+
+	/**
+	 * The `createChildDataStoreSync` method allows for the synchronous creation of a detached child data store. This is particularly
+	 * useful in scenarios where immediate availability of the child data store is required, such as during the initialization
+	 * of a parent data store, or when creation is in response to synchronous user input.
+	 *
+	 * In order for this function to succeed:
+	 * 1. The parent data store's factory must also be an `IFluidDataStoreRegistry`.
```

Comment:
Should we remove # 1? It's already a requirement to be able to create child data store, so it's not unique to sync path.
# 2 seems like also the existing requirement (while slightly different layer, but PureDataObjectFactory.createChildInstance() has same restriction.

Reply:
I don't think it's clearly documented elsewhere, and i prefer to have comments a developer could use to understand the whole problem, if for some reason it is not working.